### DISCUSSION TITLE
expr column in ColumnsSheet is hidden, or how to config default columns in built-in sheets

### DISCUSSION THREAD

**Discussion by betamint** [saulpw/visidata#2445] [Q&A] [answered]
The `expr` column in `ColumnsSheet` (appeared after press `Shift+C`) used to be displayed by default, but is hidden on v3.0.2.

When I need to view or edit expressions, I have to do the following tedious procedure each time. I think I must be missing something here.

1. Press `Shift+C` to open `ColumnsSheet`; 
2. Press `Shift+C` again to open `ColumnsSheet` of the `ColumnsSheet`;
3. Edit the width of `expr` to unhide it.

How to change `expr`'s default width to unhide it in `ColumnsSheet`?

Or, is there anything general to edit defaults for built-in sheets like `ColumnsSheet`?

**Comment by saulpw** [comment, marked as answer]
Hi @betamint,

If you want it to be exposed by default, you can add this to your .visidatarc:

    ColumnsSheet.knownCols.expr.width = 10

This will work for any sheet and column too, not just internal sheets.

I'll take this as a vote to make the `expr` unhidden by default again, which I'm not opposed to.  We hid the `expr` column on the ColumnsSheet by default because it seemed like it was internal clutter.

**Reply by betamint** [reply]
It seems to me that an editable `expr` column is a must if one wants to use expressions. Expressions dropped from a major feature to something I avoid to use when expressions cannot be easily edited.

**Reply by saulpw** [reply]
Well then how about an `edit-expr` command that will allow you to edit the expression of the current column?

    Sheet.addCommand('', 'edit-expr', 'cursorCol.expr=inputExpr("edit expr: ", value=cursorCol.expr)', 'edit expression for current column')

This is a matter of workflow taste; I use expressions all the time and hardly ever edit them--I just delete/hide the column and create another one, up-arrow to get history.  But if you want to edit the expressions, let's make that as easy as possible for you.  I mean this is VisiData, you're minutes away from the workflow you want.

**Reply by betamint** [reply]
That's definitely better.

> I mean this is VisiData, you're minutes away from the workflow you want.

This just made my day :)
